### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by maximmi**
## What
* Allow to create `Codefresh` repos and pipelines from template for a collection of github repos

## Why
* We are going to move off of Travis CI for terraform modules (there are a lot of them)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export HELP_FILTER ?= help|init|codefresh/sync
+export DOCKER_NETWORK ?= default
+export BUILD_HARNESS_BRANCH ?= 0.15.3
+-include $(shell curl -sSL -o .build-harness "https://git.io/build-harness"; echo .build-harness)
+
+## Project name
+PROJECT ?= cloudposse-pipelines
+
+## Repos to sync pipelines
+REPOSITORIES ?= cloudposse/terraform-aws-s3-bucket \
```
Comment (osterman) [inline Makefile]:
```suggestion
REPOSITORIES ?= $(shell cat repositories.txt)```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ 				cloudposse/terraform-aws-mq-broker
```
Comment (osterman) [inline Makefile]:
```suggestion
```

File: pipelines_codefresh.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+version: '1.0'
```
Comment (osterman) [inline pipelines_codefresh.yml]: Move this file to `codefresh/terraform.yml`

**Comment by osterman** [comment]
[Codefresh Pipeline Automation](https://trello.com/c/j3pVgvTt/40-codefresh-pipeline-automation)

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export DEFAULT_HELP_TARGET ?= help
+export HELP_FILTER ?= help|init|codefresh/sync
+export DOCKER_NETWORK ?= default
+export BUILD_HARNESS_BRANCH ?= 0.15.3
```
Comment (aknysh) [inline Makefile:3]: why do we need `BUILD_HARNESS_BRANCH`?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+PROJECT ?= cloudposse-pipelines
+
+## Repos to sync pipelines
+REPOSITORIES ?= $(shell cat repositories.txt)
+
+## Codefresh accounts to operate with
+ACCOUNTS ?= cloudposse
+
+## Pipelines for `cloudposse`
+CLOUDPOSSE_PIPELINES ?= lint
```
Comment (aknysh) [inline Makefile:16]: not in use?

File: README.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+name: Codefresh pipelines
+
+# License of this project
+license: "APACHE2"
+
+# Copyrights
+copyrights:
+  - name: "Cloud Posse, LLC"
+    url: "https://cloudposse.com"
+    year: "2018"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    year: "2019"
```

File: README.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+# Canonical GitHub repo 
+github_repo: cloudposse/codefresh-pipelines
+
+# Badges to display
+badges:
+  - name: "Build Status"
+    image: "https://travis-ci.org/cloudposse/build-harness.svg?branch=master"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    image: "https://travis-ci.org/cloudposse/codefresh.svg?branch=master"
```

File: README.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    url: "https://travis-ci.org/cloudposse/build-harness"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    url: "https://travis-ci.org/cloudposse/codefresh"
```

File: README.yaml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+# References
+#references:
+#TODO: add all terraform modules here
+#  - name: "Wikipedia - Test Harness"
+#    description: 'The `build-harness` is similar in concept to a "Test Harness"'
+#    url: "https://en.wikipedia.org/wiki/Test_harness"
+
+# Short description of this project
+description: |-
+  This `codefresh-pipelines` repo allows to create and update `Codefresh` pipelines for a collection of terraform modules and other repos.
```
Comment (aknysh) [inline README.yaml]:
```suggestion
  This `codefresh` repo allows to create and update `Codefresh` pipelines for a collection of terraform modules and other repos.
```

File: README.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+# How to use this project
+usage: |-
+  If you need to add new repo, just update `Makefile` adding new repo to `REPOSITORIES` variable. Make a PR and as soon as it will be closed, new `Codefresh` repository will be created with proper pipeline.
```
Comment (aknysh) [inline README.yaml]:
```suggestion
  If you need to add new repo, just update `Makefile` adding new repo to `REPOSITORIES` variable. Open a PR and as soon as it's closed, a new `Codefresh` repository will be created with proper pipeline.
```

File: README.yaml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  If you need to update pipelines, then feel free to update proper files in `templates` folder, make a PR and as soon as it will be closed all supported pipelines will be updated.
```
Comment (aknysh) [inline README.yaml]:
```suggestion
  If you need to update pipelines, feel free to update the files in `templates` folder, open a PR and as soon as it's closed all supported pipelines will be updated.
```

**Review by aknysh — COMMENTED** [review]
see comments

File: README.yaml (hunk 7 above)
Comment (osterman) [reply README.yaml]: Should not be a travis badge =)

File: Makefile (hunk 1 above)
Comment (maximmi) [reply Makefile]: done

File: Makefile (hunk 2 above)
Comment (maximmi) [reply Makefile]: done

File: pipelines_codefresh.yml (hunk 3 above)
Comment (maximmi) [reply pipelines_codefresh.yml]: done

File: Makefile (hunk 4 above)
Comment (maximmi) [reply Makefile:3]: this is how we pin build_harness version and here we need at least this version

File: Makefile (hunk 5 above)
Comment (maximmi) [reply Makefile:16]: it is in use. Within `templates` folder we are looking for `account` from this list

File: README.yaml (hunk 6 above)
Comment (maximmi) [reply README.yaml]: done

File: README.yaml (hunk 7 above)
Comment (maximmi) [reply README.yaml]: fixed

File: README.yaml (hunk 9 above)
Comment (maximmi) [reply README.yaml]: fixed

File: README.yaml (hunk 11 above)
Comment (maximmi) [reply README.yaml]: done

File: README.yaml (hunk 10 above)
Comment (maximmi) [reply README.yaml]: done

File: README.yaml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    year: "2019"
+
+# Canonical GitHub repo 
+github_repo: cloudposse/codefresh-pipelines
+
+# Badges to display
+badges:
+  - name: "Build Status"
+    image: "https://g.codefresh.io/api/badges/pipeline/cloudposse/cloudposse%2Fcodefresh%2Fcodefresh?type=cf-1"
+    url: "https://g.codefresh.io/public/accounts/cloudposse/pipelines/cloudposse/codefresh/codefresh"
```
Comment (osterman) [inline README.yaml:20]: Missing release badge

File: README.yaml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+related:
+  - name: "build-harness"
+    description: "Collection of Makefiles to facilitate building Golang projects, Dockerfiles, Helm charts, and more"
+    url: "https://github.com/cloudposse/build-harness"
+  - name: "Packages"
+    description: "Cloud Posse installer and distribution of native apps"
+    url: "https://github.com/cloudposse/packages"
+
+# References
```
Comment (osterman) [inline README.yaml]: Let's remove this commented block.

File: codefresh.yml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+# Build a service with environment variables
+version: '1.0'
+
+steps:
+  apply_codefresh_sync:
+    title: Sync codefresh pipelines
+    image: cloudposse/build-harness:${{BUILD_HARNESS_VERSION}}
+    commands:
+#    Requires CF_API_KEY, here it will get it from Shared Configuration
```
Comment (osterman) [inline codefresh.yml]: Awkward formatting

image

File: templates/cloudposse/lint.yaml [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+      repo: {{ (ds "repository") }}
+      type: git
+  contexts:
+    - terraform modules
+  variables: []
+  specTemplate:
+    location: url
+#TODO: update when it will be merged to master and create a short link
+    url: >-
+      https://raw.githubusercontent.com/cloudposse/codefresh/initial-implementation/codefresh/terraform.yml
```
Comment (osterman) [inline templates/cloudposse/lint.yaml]:
```suggestion
      https://raw.githubusercontent.com/cloudposse/codefresh/master/codefresh/terraform.yml
```

File: README.yaml (hunk 13 above)
Comment (maximmi) [reply README.yaml]: done

File: README.yaml (hunk 12 above)
Comment (maximmi) [reply README.yaml:20]: done

File: codefresh.yml (hunk 14 above)
Comment (maximmi) [reply codefresh.yml]: done

File: templates/cloudposse/lint.yaml (hunk 15 above)
Comment (maximmi) [reply templates/cloudposse/lint.yaml]: done

**Review by osterman — APPROVED** [review]

**Pull request merged**